src/commands/add.js

```diff
@@ -0,0 +1,208 @@
+const { loadAbiFromEtherscan, loadAbiFromBlockScout } = require('./init')
+const EthereumABI = require('../protocols/ethereum/abi')
+const { fixParameters } = require('../command-helpers/gluegun')
+
+const HELP = `
+${chalk.bold('graph add')} <address> [<subgraph-manifest default: "./subgraph.yaml">]
+
+${chalk.dim('Options:')}
+
+      --abi <path>              Path to the contract ABI (default: download from Etherscan)
+      --contract-name           Name of the contract (default: Contract)
+      --merge-entities          Whether to merge entities with the same name (default: false)
+  -h, --help                    Show usage information
+`
+
+module.exports = {
+  description: 'Creates a new subgraph with basic scaffolding',
+  run: async toolbox => {
+    // Obtain tools
+    let { print, system } = toolbox
+
+    // Read CLI parameters
+    let {
+      abi,
+      contractName,
+      h,
+      help,
+      mergeEntities
+    } = toolbox.parameters.options
+
+    let address = toolbox.parameters.first
+    let manifestPath = toolbox.parameters.second ? toolbox.parameters.second : './subgraph.yaml'
+    contractName = contractName ? contractName : 'Contract'
+
+    // Validate the address
+    if (!address) {
+      print.error('No contract address provided')
+      process.exitCode = 1
+      return
+    }
+
+    try {
+      fixParameters(toolbox.parameters, {
+        h,
+        help,
+        mergeEntities,
+      })
+    } catch (e) {
+      print.error(e.message)
+      process.exitCode = 1
+      return
+    }
+
+    // Show help text if requested
+    if (help || h) {
+      print.info(HELP)
+      return
+    }
+
+    const dataSourcesAndTemplates = await DataSourcesExtractor.fromFilePath(manifestPath)
+    let protocol = Protocol.fromDataSources(dataSourcesAndTemplates)
+    let manifest = await Subgraph.load(manifestPath, {protocol: protocol})
+    let network = manifest.result.get('dataSources').get(0).get('network')
+    let result = manifest.result.asMutable()
+
+    let entities = getEntities(manifest)
+    let contractNames = getContractNames(manifest)
+    if (contractNames.indexOf(contractName) !== -1) {
+      print.error(
+        `Datasource or template with name ${contractName} already exists, please choose a different name`,
+      )
+      process.exitCode = 1
+      return
+    }
+
+    let ethabi = null
+    if (abi) {
+      ethabi = EthereumABI.load(contractName, abi)
+    } else {
+      if (network === 'poa-core') {
+        ethabi = await loadAbiFromBlockScout(EthereumABI, network, address)
+      } else {
+        ethabi = await loadAbiFromEtherscan(EthereumABI, network, address)
+      }
+    }
+
+    let updateResult = updateEventNamesOnCollision(ethabi, entities, contractName, mergeEntities)
+    let collisionEntities = updateResult.collisionEntities
+    let onlyCollisions = updateResult.onlyCollisions
+    ethabi.data = updateResult.abiData
```

Comment:
You could do a destructuring here :)

```suggestion
    let { collisionEntities, onlyCollisions, abiData } = updateEventNamesOnCollision(ethabi, entities, contractName, mergeEntities)
    ethabi.data = abiData
```